### PR TITLE
[FIX] web: Expose datepicker configuration to view declarations

### PR DESCRIPTION
**Pull request by yajo**
The lower-level widget itself supports passing options to the underlying datepicker library, but this is not exposed to higher-level view declaration.

With this patch, one can declare a field in a view such as:

```xml
<field name="datefield" options='{"datepicker": {"daysOfWeekDisabled": [0, 6]}}'/>
```

That option would land in the widget and disable those weekdays.

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr
@Tecnativa @ged-odoo

### CODE REVIEW

**Comment by tde-banana-odoo** [comment]
Hello,

Changes look goot to me, except that I would make them in master as this is not really a bugfix. Just to be sure I would ask for @ged-odoo review.

**Comment by Yajo** [comment]
Weird, I can't imagine how this change is breaking the test... :confused:

File: addons/web/static/src/js/fields/basic_fields.js [hunk 1]
```diff
@@ -468,7 +468,16 @@ var FieldDate = InputField.extend({
      * @private
      */
     _makeDatePicker: function () {
-        return new datepicker.DateWidget(this, {defaultDate: this.value});
+        var datepicker_options;
+        try {
+            datepicker_options = this.attrs.options.datepicker;
```
Comment (tde-banana-odoo) [inline addons/web/static/src/js/fields/basic_fields.js]: Isn't it simpler to do datepicker_options = this.attrs.options.datepicker || {} ?

Comment (Yajo) [reply addons/web/static/src/js/fields/basic_fields.js]: Hmm it must be that.

Comment (ged-odoo) [reply addons/web/static/src/js/fields/basic_fields.js]: tde is right

Comment (Yajo) [reply addons/web/static/src/js/fields/basic_fields.js]: I was worried about `this.attrs.options` being `undefined`, but I didn't know about `this.nodeOptions`. It's changed now.

**Comment by Yajo** [comment]
Docs added

**Comment by tde-banana-odoo** [comment]
Landed in 11 at https://github.com/odoo/odoo/commit/71334b13bd31c7f5a0c1704584b28274785f95e3 . Thanks to everybody !

**Pull request closed without merging**